### ISSUE TITLE
Bad error message when bad command line arguments are supplied

### ISSUE THREAD

**Issue by mg03** [hashicorp/consul#1701]
Hi Using consul on ubtuntu 14.04

Config file is 
{
  "datacenter": "tpa",
  "bind_addr": "10.10.32.5",
  "advertise_addr": "10.10.32.5",
  "node_name": "vm-10-10-32-5",
  "rejoin_after_leave": true,
  "domain": "tpa",
  "retry_join": [ 10.10.32.5, 10.10.33.5, 10.10.34.5 ],
  "server": true,
  "bootstrap_expect": 3,
  "encrypt": "cg8StVXbQJ0gPvMd9o7yrg==",
  "data_dir": "/var/lib/consul",
  "ui_dir": "/usr/share/consul-ui/dist",
  "disable_remote_exec": true
}

consul version
Consul v0.6.3
Consul Protocol: 3 (Understands back to: 1)

consul agent client 0.0.0.0 -data-dir=/var/lib/consul -config-dir=/etc/consul
==> Must specify data directory using -data-dir

Why do I keep getting the message to provide -data-dir even when I am providing it both in the config fil and on the command line?

**Comment by highlyunavailable** [comment]
This is a bad error message that I will leave to the Hashicorp guys to resolve, but your extraneous command line parameters are breaking it:
consul agent __client 0.0.0.0__ -data-dir=/var/lib/consul -config-dir=/etc/consul

Remove the bolded part and it will work fine. Also if you're specifying it in the file, you don't need to specify it in the command line. Just run `consul agent -config-dir=/etc/consul` and it should work fine.

**Comment by slackpad** [comment]
Thanks @highlyunavailable - agree the error message is super confusing here.

**Comment by mg03** [comment]
@highlyunavailable , @slackpad : thank you issue resolved

**Comment by slackpad** [comment]
I'll leave this open as a reminder to take a look at improving the error message.

**Comment by jkirschner-hashicorp** [comment]
It is my understanding that this was fixed by #7762, specifically [in this line](https://github.com/hashicorp/consul/blob/e41d6ee60f072c9b07c858a65ae1c02a3d7ac5ab/command/agent/agent.go#L157).

The error message now looks like:
```
==> Unexpected extra arguments: [client 0.0.0.0 -config-dir ./config/]
```

**Comment by alfonsovgs** [comment]
> It is my understanding that this was fixed by #7762, specifically [in this line](https://github.com/hashicorp/consul/blob/e41d6ee60f072c9b07c858a65ae1c02a3d7ac5ab/command/agent/agent.go#L157).
> 
> The error message now looks like:
> 
> ```
> ==> Unexpected extra arguments: [client 0.0.0.0 -config-dir ./config/]
> ```

how to resolve the error client 0.0.0.0? I'm using docker
